syntax = "proto3";  // 声明protobuf的版本

package fixbug; // 声明了代码所在的包（对于C++来说就是namespace)

// 定义下面的选项，表示生成service服务类和rpc方法描述，默认不生成
option cc_generic_services = true;

// 定义错误信息
message ResultCode{
    int32 errcode = 1;
    bytes errmsg = 2;
}
// 定义登录请求消息类型 name pwd
message LoginRequest{
    bytes name = 1;
    bytes pwd = 2;
}

// 定义登录响应消息类型
message LoginResponse{
    ResultCode result = 1;
    bool success = 2;
}

// 定义请求好友列表类型
message GetFriendListsRequest{
    uint32 userid = 1;
}

message User{
    bytes name = 1;
    uint32 age = 2;
    enum Sex{
        MAN = 0;
        WOMAN = 1;
    }
    Sex se = 3;
}

// 定义好友返回类型
message GetFriendListsResponce{
    ResultCode result = 1;
    repeated User friend_list = 2;    // 列表
}

//在前面protobuf中 定义了对象,接下来要在protobuf 中,定义相关的函数

// 在protobuf里面怎么定义描述rpc方法的描述 - service
service UserServiceRpc{
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc GetFriendLists(GetFriendListsRequest) returns(GetFriendListsResponce);
}